song_player/songplayer.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <QWidget>
+#include <QMediaPlayer>
+#include <QAbstractButton>
+#include <QBoxLayout>
+#include <QStandardPaths>
+#include <QFileDialog>
+#include <QFileInfo>
+#include <QShortcut>
+#include <QCoreApplication>
+#include <QToolButton>
+#include <QTime>
+#include <QSlider>
+#include <QLabel>
+#include <QMouseEvent>
+#include <QStyle>
+
+QT_FORWARD_DECLARE_CLASS(QLabel)
+QT_FORWARD_DECLARE_CLASS(QSlider)
```

Comment:
I've never seen `QT_FORWARD_DECLARE_CLASS()` used before and nothing on Google is helping out....can you, @enochtsang or @bseto chime in?